### ISSUE TITLE
"integer | null" outputs the wrong schema

### ISSUE THREAD

**Issue by puppybits** [YousefED/typescript-json-schema#356]
We want to define a property as an integer or null. But using the @TJS-integer to overrides the type instead of only replacing the number part of the definition.

This TS
```ts
interface MyObject {
  /**
   * @TJS-type integer
   */
  intOrNull: number | null;
}
```

Outputs this:
```ts
{
  "MyObject": {
    "type": "object",
    "properties": {
        "intOrNull": {
            "type": integer
        }
    }
  }
}
```

But the output should be:
```ts
{
  "MyObject": {
    "type": "object",
    "properties": {
        "intOrNull": {
            "type": [
                  "integer",
                  "null"
              ]
        }
    }
  }
}
```

**Comment by domoritz** [comment]
Are you using strict mode in tsconfig? What's your tsconfig?

**Comment by jomido** [comment]
I encountered this when going the programmatic route. I found that I must use `strict: true` when constructing the `TJS.Program`:

```typescript
const program = TJS.getProgramFromFiles(
      [fileName],
      {
        // ...
        strict: true,
      }
    )
```

If I omit `strict: true`, then fields typed as `<someType> | null` have the incomplete schema mentioned by @puppybits:`<someType>` is a member of the type array, but `null` is missing.

If I include `strict: true`, then fields typed as `<someType | null` have complete schemas: both `<someType>` and `null` are members of the type array.
